plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/LegendAction.java

```diff
@@ -45,12 +45,8 @@
 	public LegendAction(GraphicalViewerImpl viewer, IGraphicalFeatureModel featureModel) {
 		super();
 		this.featureModel = featureModel;
-		if (!featureModel.isLegendHidden()) {
-			setText(HIDE_LEGEND);
-		} else {
-			setText(SHOW_LEGEND);
-		}
-
+		setText(SHOW_LEGEND);
+		setChecked(!featureModel.isLegendHidden());
```

Comment:
The text is fixed although the check state is not. Why?

Reply:
That's intended.